trio/tests/test_path.py

```diff
@@ -3,30 +3,44 @@
 import pytest
 
 import trio
-from trio._file_io._path import AsyncAutoWrapperType as Type
+from trio._path import AsyncAutoWrapperType as Type
+from trio._file_io import AsyncIOWrapper
 
 
 @pytest.fixture
 def path(tmpdir):
     p = str(tmpdir.join('test'))
-    return trio.AsyncPath(p)
+    return trio.Path(p)
 
 
 def method_pair(path, method_name):
     path = pathlib.Path(path)
-    async_path = trio.AsyncPath(path)
+    async_path = trio.Path(path)
     return getattr(path, method_name), getattr(async_path, method_name)
 
 
 async def test_open_is_async_context_manager(path):
     async with await path.open('w') as f:
-        assert isinstance(f, trio.AsyncIOWrapper)
+        assert isinstance(f, AsyncIOWrapper)
 
     assert f.closed
 
 
+async def test_cmp_magic():
+    a, b = trio.Path(''), trio.Path('')
+    assert a == b
+    assert not a != b
+
+    a, b = trio.Path('a'), trio.Path('b')
+    assert a < b
+    assert b > a
+
+    assert not a == None
```

Comment:
This would probably be a good test to use a loop (either explicitly or via pytest parametrization) to check all combinations of `pathlib.Path` and `trio.Path` objects.

We probably also want similar tests for passing different types to `trio.Path.__init__`, `trio.Path.__div__`, `trio.Path.joinpath`.